### PR TITLE
fix: request bodies should take required property into account

### LINKED ISSUE #1111: Orval does not respect the "optional by default" `requestBody` for OpenAPI 3.0
`requestBody` is optional by default in OpenAPI 3.0[[1]](https://swagger.io/docs/specification/describing-request-body/). Orval does not respect that; it also does not respect `required: false`.

[BUG REPRO REPO](https://github.com/xbc5/orval-bug-repro)

## What are the steps to reproduce this issue?

1. exclude `required: true` for a `requestBody`;
2. generate the client;

## What happens?

The request body is not optional.

## What were you expecting to happen?

The request body to be optional.

## Any logs, error output, etc?

No

## Any other comments?

The new Star Wars is overrated.

## What versions are you using?

**Operating System:** Fedora 38
**Package Version:** 6.22.1
**Browser Version:** N/A

### PR DESCRIPTION
**Pull request by AffeJonsson**
## Status

**READY**

## Description

~~Fixes #1111~~

Takes the required property of request bodies into account. We till make it required by default, but setting it to required: false will make it optional.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Changelog Entry (unreleased)

## Steps to Test or Reproduce

1. yarn generate:swr
2. check tests/generated/swr/optional-request-body/endpoints.ts. Endpoints without a required property has the request body required. Endpoints with a required: false property has the request body as optional.

### CODE REVIEW

**Review by melloware — APPROVED** [review]

File: tests/specifications/optional-request-body.yaml [hunk 1]
```diff
@@ -0,0 +1,223 @@
+openapi: '3.0.0'
```
Comment (soartec-lab) [inline tests/specifications/optional-request-body.yaml:1]: We will continue to maintain the tests, so we would like to keep them to a minimum as much as possible. Is it possible to slim down to a minimized case that only checks the request body?

**Review by soartec-lab — COMMENTED** [review]
@AffeJonsson 
Hi, Thank you for great PR, I commented on one consultation, so please check it.

File: tests/specifications/optional-request-body.yaml (hunk 1 above)
Comment (AllieJonsson) [reply tests/specifications/optional-request-body.yaml:1]: Absolutely, I'll fix!

**Comment by anymaniax** [comment]
I understand that in the specification it’s optional per default but 99% if you have a request body you wanted required no?

**Comment by AffeJonsson** [comment]
That's probably true. Maybe it would be better if it's required as default. I'll change that, but at least with this change it'll take the required prop into account

File: packages/core/src/getters/body.ts [hunk 2]
```diff
@@ -61,6 +63,17 @@ export const getBody = ({
...
       es5IdentifierName: true,
     });
+    if (isReference(requestBody)) {
+      const { schema: bodySchema } = resolveRef<RequestBodyObject>(
+        requestBody,
+        context,
+      );
+      if (bodySchema.required !== undefined) {
+        isOptional = !bodySchema.required;
+      }
```
Comment (anymaniax) [inline packages/core/src/getters/body.ts:73]: What is the value of required if we don’t set it?

**Review by anymaniax — APPROVED** [review]

**Review by soartec-lab — APPROVED** [review]
Thank you.

**Pull request merged**